test/unit/sourceset.test.js

```diff
@@ -0,0 +1,479 @@
+/* eslint-env qunit */
+import videojs from '../../src/js/video.js';
+import document from 'global/document';
+import window from 'global/window';
+import log from '../../src/js/utils/log.js';
+import sinon from 'sinon';
+
+const Html5 = videojs.getTech('Html5');
+const wait = 1;
+let qunitFn = 'module';
+
+if (!Html5.canOverrideAttributes()) {
+  qunitFn = 'skip';
+}
+
+const oldMovingMedia = Html5.prototype.movingMediaElementInDOM;
+const validateSource = function(assert, player, sources, checkMediaElSource = true) {
+  const tech = player.tech_;
+  const mediaEl = tech.el();
+
+  if (checkMediaElSource) {
+    assert.equal(mediaEl.src, sources[0].src, 'mediaEl.src is correct');
+    assert.equal(mediaEl.getAttribute('src'), sources[0].src, 'mediaEl attribute is correct');
+    assert.equal(tech.src(), sources[0].src, 'tech is correct');
+  }
+};
+
+const setupEnv = function(env, testName) {
+  env.fixture = document.getElementById('qunit-fixture');
+
+  if (testName === 'change video el' || testName === 'change audio el') {
+    Html5.prototype.movingMediaElementInDOM = false;
+  }
+
+  env.sourcesets = 0;
+  env.hook = (player) => player.on('sourceset', () => env.sourcesets++);
+  videojs.hook('setup', env.hook);
+
+  if ((/audio/i).test(testName)) {
```

Comment:
good to test creating with `audio` element but there's no need to set mp3 urls, plus the mp3 urls were creating a lot of 404 errors in the console, so, just using a single mp4 test source and only mp4 sources